### PR TITLE
[FRONT-109] Fetch stream activity status through StreamrClient

### PR DESCRIPTION
**Pull request by mondoreale**
In this PR I replace fetching stream status via e&e with fetching it via ~~`streamr-client`~~ the API. Notes:

- `useLastMessageTimestamp` hook lets you determine the latest message timestamp.
- `getStreamActivityStatus` utility turns a timestamp in a proper `StatusIcon` instance.
- I extracted `ClientProvider` from `PreviewView` and wrapped everything in `StreamPage/index.jsx` with it (both `View` and `Edit` components) – see 773bb2c.
- Wrote a couple of new `cypress` commands so that you can, for example, publish a message to a stream (it waits for it to complete).

And a bunch of other things. Hope you guys like it. Cheers!

### CODE REVIEW

File: app/src/shared/hooks/useLastMessageTimestamp.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    }, [])
+
+    const refreshKeyRef = useRef(refreshKey)
+
+    useEffect(() => {
+        refreshKeyRef.current = refreshKey
+
+        const attach = async () => {
+            try {
+                await client.resend({
```
Comment (juhah) [inline app/src/shared/hooks/useLastMessageTimestamp.js:62]: There is some weirdness here with the `client` that it keeps updating if the resend was not successful, which in turn runs the effect again. You can try it by setting the params as:

```js
await client.resend({
    stream: `invalid-${streamId}`,
    ....
})
```

The loading will go crazy:
ezgif-2-87c10eca867a

See also https://linear.app/streamr/issue/FRONT-113/stream-client-websocket-connection-keeps-failing-after-automatic

Comment (mondoreale) [reply app/src/shared/hooks/useLastMessageTimestamp.js:62]: Haha, what a gem! Thanks @juhah for finding that one. [`FRONT-113`](https://linear.app/streamr/issue/FRONT-113/stream-client-websocket-connection-keeps-failing-after-automatic) is next on my list!

**Review by juhah — CHANGES_REQUESTED** [review]
Otherwise everything looks good but the update issue with the invalid id / refreshing client should be solved. It might have some real-world effect if the stream fetch fails for some other reason. Requesting changes here but feel free to merge if it looks like the problem is actually with the client provider 👍

**Comment by mondoreale** [comment]
Rejection is fine, sir. There's more to the bug than this. Apparently even with `autoDisconnect: false` when a resend request fails the client gets disconnected "hard" causing ClientProvider to recreate `client` instance, thus causing that `useEffect` to be picked up.

Looking into it!

**Comment by mondoreale** [comment]
Fetching the last message using the API directly. No streamr client involved. Let's move on.

File: app/src/shared/hooks/useLastMessageTimestamp.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            })
+            isFetchingRef.current = true
+        }
+    }, [])
+
+    useEffect(() => {
+        const attach = async () => {
+            try {
+                const [message] = await get({
+                    url: routes.api.streams.lastMessage({
```
Comment (juhah) [inline app/src/shared/hooks/useLastMessageTimestamp.js:65]: there is a getter for `getStreamData` in `modules/stream/services` that uses almost similar route, `routes.api.streams.data`. Might be an idea to combine those, though it's good like this as well

File: app/src/routes/definitions.json [hunk 3]
```diff
@@ -152,7 +151,8 @@
             "permissions": {
                 "index": "<api>/streams/:streamId/permissions",
                 "show": "<api>/streams/:streamId/permissions/:id"
-            }
+            },
+            "lastMessage": "<api>/streams/:streamId/data/partitions/:partition/last"
```
Comment (juhah) [inline app/src/routes/definitions.json:155]: This is quite similar to the `data` entry, see comment below

**Review by juhah — APPROVED** [review]
Working great 👍

**Comment by mondoreale** [comment]
@juhah I put routes closer to each other but passed on defining a combined utility or another one for "last" messages – less moving parts for something that will, most likely, change in the nearest future (there's still the idea of using client for this).

Hope that's all fine.

@tumppi any thoughts on this PR? Otherwise I'll merge it shortly! 🚀

**Pull request merged**